modules/migration-configuring-cors-4.adoc

```diff
@@ -0,0 +1,90 @@
+// Module included in the following assemblies:
+//
+// migration/migrating_openshift_3_to_4/migrating-openshift-3-to-4.adoc
+[id='migration-configuring-cors-4_{context}']
+= Configuring cross-origin resource sharing on {product-title} 4 clusters
+
+If you are migrating to or from an {product-title} 4 cluster that does not have the CAM tool installed, you must configure configure cross-origin resource sharing so that the CAM tool can access the cluster's API server.
+
+.Procedure
+
+. Log in to the CAM tool cluster.
+. Obtain the CAM tool URL:
++
+----
+$ oc get -n openshift-migration route/migration -o go-template='(?i}//{{ .spec.host }}(:|\z){{ println }}' | sed 's,\.,\\.,g'
+----
+
+. Log in to the {product-title} 4 cluster.
+. Edit the API server CR:
++
+----
+$ oc edit authentication.operator cluster
+----
+
+. Add the CAM tool URL to the `additionalCORSAllowedOrigins` stanza:
++
+[source,yaml]
+----
+spec:
+  additionalCORSAllowedOrigins:
+  - (?i)//migration-openshift-migration\.apps\.cluster\.com(:|\z) <1>
+----
+<1> Update the CAM tool URL.
++
+[NOTE]
+====
+This example uses the following syntax:
+
+* The `(?i)` makes it case-insensitive.
+* The `//` pins to the beginning of the domain and matches the double slash
+following `http:` or `https:`.
+* The `\.` escapes dots in the domain name.
+* The `(:|\z)` matches the end of the domain name `(\z)` or a port separator
+`(:)`.
+====
+
+. Save the file to apply the changes.
+
+. Edit the Kubernetes API server CR:
++
+----
+$ oc edit kubeapiserver.operator cluster
+----
+
+. Add `corsAllowedOrigins` and the CAM tool URL to the `unsupportedConfigOverrides` stanza:
++
+[source,yaml]
+----
+spec:
+  unsupportedConfigOverrides:
+    corsAllowedOrigins:
+    - (?i)//migration-openshift-migration\.apps\.cluster\.com(:|\z) <1>
+----
+<1> Update the CAM tool URL.
+
+. Save the file to apply the changes.
+
+. Verify the configuration:
++
+----
+$ curl -v -k -X OPTIONS \
+"<cluster_url>/apis/migration.openshift.io/v1alpha1/namespaces/openshift-migration/migclusters" \ <1>
+-H "Access-Control-Request-Method: GET" \
+-H "Access-Control-Request-Headers: authorization" \
+-H "Origin: <CAM_tool_url>" <2>
```

Comment:
-H "Origin: https://<CAM_tool_url>"

Marco and myself lost time today trying to figure out why verification with not working.
Turns out we forgot to include 'http://' or 'https://' in the Origin when we attempted to verify.

The scheme is required based on how we defined the rule in master-config.yml.